airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -82,31 +113,105 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
+        normal_streams = []
+        concurrent_streams = []
+
+        for stream in streams:
+                dataset_id, table_id = stream['parent_stream'].split(".")
+                where_clause = stream["where_clause"]
+                stream_name = stream["name"]
+                if sync_method == "Standard":
+                    table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, stream_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
+                else:
+                    try:
+                        table_obj = TableChangeHistory(project_id, dataset_id, table_id, stream_name, where_clause=where_clause, fallback_start=change_history_start, slice_range=slice_range, authenticator=self._auth)
+                        next(table_obj.read_records(sync_mode=SyncMode.full_refresh))
+                    except exceptions.HTTPError as error:
+                        self.logger.warning(f"Table had no inserts in the last seven days so no change history is available causing the error: {str(error.response.text)}")
+                        table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, stream_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
+                if isinstance(table_obj, TableChangeHistory):
+                    concurrent_streams.append(table_obj.stream)
+                else:
+                    normal_streams.append(table_obj.stream)
+        for dataset in BigqueryDatasets(project_id=project_id, authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
             dataset_id = dataset.get("datasetReference")["datasetId"]
             # list and process each table under each base to generate the JSON Schema
-            for table_info in BigqueryTables(dataset_id=dataset_id, project_id=config["project_id"], authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
+            for table_info in BigqueryTables(dataset_id=dataset_id, project_id=project_id, authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
                 table_id = table_info.get("tableReference")["tableId"]
                 if sync_method == "Standard":
-                    table_obj = IncrementalQueryResult(config["project_id"], dataset_id, table_id, authenticator=self._auth)
+                    table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
                 else:
                     try:
-                        table_obj = TableChangeHistory(config["project_id"], dataset_id, table_id, authenticator=self._auth)
+                        table_obj = TableChangeHistory(project_id, dataset_id, table_id, fallback_start=change_history_start, slice_range=slice_range, authenticator=self._auth)
```

Comment:
Why do we instantiate a `TableChangeHistory` if what we use is `TableChangeHistory.stream`? Why not instantiate the `BigqueryCDCStream` directly?

Reply:
That makes sense to me. Can we document this somewhere? I think this is support helpful as a docstring for TableChangeHistory

Reply:
Ok, I will add it to the docstring of TableChangeHistory